src/search/srpso.c

```
@@ -0,0 +1,249 @@
+  cfg->c2 = (real)1.49;
+  cfg->v_max = (dpos > 0.0) ? dpos : (real)1.0;
+  cfg->seed = 0;
+}
+
+static real sr_pso_rand_span(sr_rng *rng, real span)
+{
+  const real u = (real)sr_rng_uniform01(rng);
+  return (u * (real)2.0 - (real)1.0) * span;
+}
+
+static int sr_pso_init_swarm(sr_rng *rng, int swarm, int ndim, real span,
```

Clang (14.0.6)

[warning] parameter 'rng' is unused

(misc-unused-parameters)